pkg/streamingpromql/engine_test.go

```diff
@@ -1563,15 +1586,15 @@ func TestMemoryConsumptionLimit_SingleQueries(t *testing.T) {
 					span.End()
 
 					if testCase.shouldSucceed {
-						require.NoError(t, res.Err)
-						require.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(rejectedMetrics(0)), "cortex_querier_queries_rejected_total"))
+						assert.NoError(t, res.Err)
+						assert.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(rejectedMetrics(0)), "cortex_querier_queries_rejected_total"))
 					} else {
-						require.ErrorContains(t, res.Err, globalerror.MaxEstimatedMemoryConsumptionPerQuery.Error())
-						require.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(rejectedMetrics(1)), "cortex_querier_queries_rejected_total"))
+						assert.ErrorContains(t, res.Err, globalerror.MaxEstimatedMemoryConsumptionPerQuery.Error())
+						assert.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(rejectedMetrics(1)), "cortex_querier_queries_rejected_total"))
 					}
 
 					spanStubs := spanExporter.GetSpans()
-					require.Len(t, spanStubs, 1)
+					assert.Len(t, spanStubs, 1)
```

Comment:
This one seems like it should probably stay a `require` since it will immediately panic after if the assertion doesn't hold.

Reply:
I still wanted to descend into the big memory assertions, even if the spans aren't there.